firefox-ios/Client/Frontend/Summary/LiteLLMClient.swift

```diff
@@ -0,0 +1,162 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Common
+import Shared
+
+/// A lightweight client for interacting with a chat completions endpoint.
+public class LiteLLMClient: NSObject, URLSessionDataDelegate {
+    /// NOTE: OpenAI like APIs including LiteLLM don't use websockets for streaming mode
+    /// but instead use Server Sent Events (SSE) to send chunked responses.
+    private static let sseEventDelimiter = "\n\n"
+    private static let sseDataPrefix = "data: "
+    private static let sseDoneSignal = "[DONE]"
+
+    private var logger: Logger = DefaultLogger.shared
+    private let apiKey: String
+    private let baseURL: URL
+    private lazy var session: URLSession = {
+        let config = URLSessionConfiguration.default
+        return URLSession(configuration: config, delegate: self, delegateQueue: nil)
+    }()
+
+    /// Delegate to receive streaming updates.
+    public weak var streamDelegate: LiteLLMStreamDelegate?
+
+    /// Initializes the client.
+    /// - Parameters:
+    ///   - apiKey: Your API key for authentication.
+    ///   - baseURL: Base URL of the server.
+    public init(apiKey: String, baseURL: URL) {
+        self.apiKey = apiKey
+        self.baseURL = baseURL
+    }
+
+    /// Sends a chat completion request.
+    /// - Parameters:
+    ///   - messages: Array of `ChatMessage`.
+    ///   - model: Model identifier.
+    ///   - maxTokens: Maximum tokens to generate.
```

Comment:
would love an explanation of the `1000` as well

Reply:
Definitely will add a comment here. But this has to do with budget limits.

Reply:
Updated [here](https://github.com/mozilla-mobile/firefox-ios/pull/27779/commits/b49f45a9727f06a605cdfd65ceab62325dd6626a) to use a separate struct and no defaults for now. I will add a comment at the call site when the struct is used about the 1000 limit. I think this is cleaner instead of having magic values scattered all over as I did before 😄 Let me know what you think !

Reply:
perfect! like this so much better, nice job!!